### PR TITLE
get optimizer and scheduler as arguments to train

### PR DESCRIPTION
**Pull request by deepchatterjeeligo**
It is helpful to pass in different optimizer and scheduler names are arguments, so that they can be stored and passed from config in pyproject.toml

### CODE REVIEW

File: libs/trainer/mlpe/trainer/trainer.py [hunk 1]
```diff
@@ -110,14 +110,15 @@ def train(
...
     lr: float = 1e-3,
-    min_lr: float = 1e-5,
-    decay_steps: int = 10000,
-    weight_decay: float = 0.0,
     early_stop: int = 20,
     # misc params
     device: Optional[str] = None,
     use_amp: bool = False,
     profile: bool = False,
+    optimizer_name: str = "Adam",
```
Comment (EthanMarx) [inline libs/trainer/mlpe/trainer/trainer.py:118]: Similar comments as #38. We can annotate this as 

```python
def train(
    optimizer: Callable, 
    optimizer_kwargs: Dict,
    ...
):
    optimizer = optimzer(optimizer_kwargs)
```
and then in the pyproject.toml have

```toml
[tool.typeo.scripts.train]
optimizer = "torch.optimizers.lr_scheduler.Adam"
optimizer_kwargs = {arg1: "arg1"}

Comment (deepchatterjeeligo) [reply libs/trainer/mlpe/trainer/trainer.py:118]: good idea.

Comment (deepchatterjeeligo) [reply libs/trainer/mlpe/trainer/trainer.py]: done.

**Review by EthanMarx — APPROVED** [review]
Great
